src/Data/Dwarf/Reader.hs

```diff
@@ -23,14 +26,32 @@ module Data.Dwarf.Reader
   , drTarget64
   ) where
 
-import           Data.Binary.Get
-  ( Get
+import           Data.Binary.Get  ( Get
   , getWord16be, getWord32be, getWord64be
   , getWord16le, getWord32le, getWord64le
   )
+
+import           qualified Data.Binary.Get.Internal as GI
+
 import           Data.Int (Int16, Int32, Int64)
+import           qualified Data.ByteString as B
+import           qualified Data.ByteString.Unsafe as B
+import           qualified Data.Bits as Bits
+
 import           Data.Word (Word16, Word32, Word64)
 
+word24le :: B.ByteString -> Word32
+word24le s = fromIntegral (s `B.unsafeIndex` 0) Bits..|. (fromIntegral (s `B.unsafeIndex` 1) `Bits.unsafeShiftL` 8) Bits..|. (fromIntegral (s `B.unsafeIndex` 2) `Bits.unsafeShiftL` 16)
```

Comment:
Please split this up into multiple lines for readability's sake:

```hs
word24le s =
   fromIntegral (s `B.unsafeIndex` 0) Bits..|.
  (fromIntegral (s `B.unsafeIndex` 1) `Bits.unsafeShiftL` 8) Bits..|.
  (fromIntegral (s `B.unsafeIndex` 2) `Bits.unsafeShiftL` 16)
```

And similarly for `word24be`.